hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/TargetRepository.java

```diff
@@ -177,11 +196,32 @@ void setAssignedAndInstalledDistributionSetAndUpdateStatus(@Param("status") Targ
      */
     Page<Target> findByAssignedDistributionSetId(Pageable pageable, Long setID);
 
+    /**
+     * Select controllerIds which do have a ceratain {@linkplain DistributionSet} assigned
+     * @param dsId the id of the {@linkplain DistributionSet}
+     * @return a list of controller-ids which do have the specified {@linkplain DistributionSet} assigned.
+     */
+    @Query("SELECT t.controllerId FROM JpaTarget t where t.assignedDistributionSet.id = :dsId")
+    Set<String> findControllerIdsByAssignedDistributionSet(Long dsId);
+
+    /**
+     * Finds all target-controller IDs that have a {@link DistributionSet}
```

Comment:
Better: Finds all controllerIds which aren't assigned to the given {@link DistributionSet}